Regression in v0.11.1: panic: unaligned 64-bit atomic operation

**Kopia-CLI**: v0.11.1

Cross-compiled `kopia` on Linux x86_64 for FreeBSD architecture 386 with following script: 

```
#!/bin/sh

export KOPIA_VERSION=$(git describe --abbrev=0)
export KOPIA_VERSION_NO_PREFIX=$(echo $KOPIA_VERSION | sed 's/^[^0-9\.]*//g')
BUILD_USER=root
BUILD_HOST=localhost

export CGO_ENABLED=0   # make sure to compile static

env GOOS=freebsd      \
    GOARCH=386         \
      go build          \
        -v               \
        -a                \
        -trimpath          \
        -o kopia.freebsd.386.$KOPIA_VERSION  \
    -ldflags "-s -w -X github.com/kopia/kopia/repo.BuildVersion=${KOPIA_VERSION_NO_PREFIX} -X github.com/kopia/kopia/repo.BuildInfo=$(git rev-parse HEAD) -X github.com/kopia/kopia/repo.BuildGitHubRepo=${BUILD_USER}@${BUILD_HOST}"
```

Previous versions 0.10.6 and 0.10.7 worked without any problem, but starting with v0.11.1, attempt to snapshot failed with following errors:

```
2022-07-06T00:22:17.098724-04:00  Snapshotting root@mx72:/ ...
panic: unaligned 64-bit atomic operation

goroutine 168 [running]:
runtime/internal/atomic.panicUnaligned()
	runtime/internal/atomic/unaligned.go:8 +0x27
runtime/internal/atomic.Xadd64(0x3a0a4e64, 0x1)
	runtime/internal/atomic/atomic_386.s:125 +0x11
github.com/golang-design/lockfree.(*Stack).Push(0x3a0a4e60, {0x8e51d20, 0x3a0bf840})
	github.com/golang-design/lockfree@v0.0.1/stack.go:45 +0xc4
github.com/kopia/kopia/internal/freepool.(*Pool).Return(0x3a0a4e58, {0x8e51d20, 0x3a0bf840})
	github.com/kopia/kopia/internal/freepool/freepool.go:30 +0x48
github.com/kopia/kopia/fs/localfs.(*filesystemDirectory).Close(0x3a0bf840)
	github.com/kopia/kopia/fs/localfs/local_fs_pool.go:72 +0x2f
github.com/kopia/kopia/snapshot/snapshotfs.estimate.func1({0x9525b9c, 0x3a074840}, {0x9528e8c, 0x3a0a4bb8})
	github.com/kopia/kopia/snapshot/snapshotfs/estimate.go:140 +0x32e
github.com/kopia/kopia/fs/ignorefs.(*ignoreDirectory).IterateEntries.func1({0x9525b9c, 0x3a074840}, {0x9528e8c, 0x3a0a4ba0})
	github.com/kopia/kopia/fs/ignorefs/ignorefs.go:163 +0x7e
github.com/kopia/kopia/fs/ignorefs.(*ignoreDirectory).IterateEntries.func1({0x9525b9c, 0x3a074840}, {0x9528ec4, 0x3a0bf840})
	github.com/kopia/kopia/fs/ignorefs/ignorefs.go:163 +0x7e
github.com/kopia/kopia/fs/localfs.(*filesystemDirectory).IterateEntries(0x3a43a900, {0x9525b9c, 0x3a074840}, 0x3a44ca20)
	github.com/kopia/kopia/fs/localfs/local_fs.go:174 +0x348
github.com/kopia/kopia/fs/ignorefs.(*ignoreDirectory).IterateEntries(0x3a0a49c0, {0x9525b9c, 0x3a074840}, 0x3a41be70)
	github.com/kopia/kopia/fs/ignorefs/ignorefs.go:161 +0x111
github.com/kopia/kopia/fs/ignorefs.(*ignoreDirectory).IterateEntries(0x3a0a49f0, {0x9525b9c, 0x3a074840}, 0x3a28f9a0)
	github.com/kopia/kopia/fs/ignorefs/ignorefs.go:161 +0x111
github.com/kopia/kopia/snapshot/snapshotfs.estimate({0x9525b9c, 0x3a074840}, {0x8e8e9aa, 0x1}, {0x9528e8c, 0x3a0a49f0}, 0x3a00b180, 0x3a03fd70, {0x3a1240c0, 0xe, ...}, ...)
	github.com/kopia/kopia/snapshot/snapshotfs/estimate.go:133 +0x31f
github.com/kopia/kopia/snapshot/snapshotfs.Estimate({0x9525b9c, 0x3a074840}, {0x9529c68, 0x3a0a49c0}, 0x3a00b180, {0x9524d88, 0x3a2310b0}, 0x1)
	github.com/kopia/kopia/snapshot/snapshotfs/estimate.go:106 +0x9df
github.com/kopia/kopia/snapshot/snapshotfs.(*Uploader).scanDirectory(0x3a3b6850, {0x9525b9c, 0x3a074840}, {0x9529c68, 0x3a0a49c0}, 0x3a00b180)
	github.com/kopia/kopia/snapshot/snapshotfs/upload_scan.go:39 +0xa1
github.com/kopia/kopia/snapshot/snapshotfs.(*Uploader).Upload.func1()
	github.com/kopia/kopia/snapshot/snapshotfs/upload.go:1231 +0xf7
created by github.com/kopia/kopia/snapshot/snapshotfs.(*Uploader).Upload
	github.com/kopia/kopia/snapshot/snapshotfs/upload.go:1226 +0xb99

```

--- a/go.mod
+++ b/go.mod
@@ -106,7 +106,6 @@ require (
 	github.com/chromedp/cdproto v0.0.0-20220515234810-83d799542a04
 	github.com/chromedp/chromedp v0.8.2
 	github.com/edsrzf/mmap-go v1.1.0
-	github.com/golang-design/lockfree v0.0.1
 	github.com/hashicorp/golang-lru v0.5.4
 	go.opentelemetry.io/otel v1.7.0
 	go.opentelemetry.io/otel/exporters/jaeger v1.7.0
@@ -118,7 +117,6 @@ require (
 	cloud.google.com/go/compute v1.7.0 // indirect
 	cloud.google.com/go/iam v0.3.0 // indirect
 	github.com/alessio/shellescape v1.4.1 // indirect
-	github.com/changkun/lockfree v0.0.1 // indirect
 	github.com/chromedp/sysutil v1.0.0 // indirect
 	github.com/go-logr/logr v1.2.3 // indirect
 	github.com/go-logr/stdr v1.2.2 // indirect

--- a/internal/freepool/freepool.go
+++ b/internal/freepool/freepool.go
@@ -2,38 +2,33 @@
 package freepool
 
 import (
-	"github.com/golang-design/lockfree"
+	"sync"
 )
 
 // Pool is a small pool of recently returned objects.
 // Unlike sync.Pool, the pool is not subject to gargbage collection under memory pressure.
 type Pool struct {
-	makeNew func() interface{}
-	clean   func(v interface{})
-	stack   lockfree.Stack
+	clean func(v interface{})
+	pool  sync.Pool
 }
 
 // Take returns an item from the pool, and if not available makes a new one.
 func (p *Pool) Take() interface{} {
-	v := p.stack.Pop()
-	if v == nil {
-		return p.makeNew()
-	}
-
-	return v
+	return p.pool.Get()
 }
 
 // Return returns an item to the pool after cleaning it.
 func (p *Pool) Return(v interface{}) {
 	p.clean(v)
-
-	p.stack.Push(v)
+	p.pool.Put(v)
 }
 
 // New returns a new free pool.
 func New(makeNew func() interface{}, clean func(v interface{})) *Pool {
 	return &Pool{
-		makeNew: makeNew,
-		clean:   clean,
+		clean: clean,
+		pool: sync.Pool{
+			New: makeNew,
+		},
 	}
 }

